edifice/runner.py

```diff
@@ -100,12 +101,9 @@ def runner():
 
     observer = Observer()
 
-    parts = list(os.path.split(args.main_file))
-    if parts[-1].endswith(".py"):
-        parts[-1] = parts[-1][:-3]
-
-    module_name = ".".join(parts)
-    main_module = importlib.import_module(module_name)
+    main_file = Path((args.main_file))
+    
+    main_module = importlib.import_module(main_file.stem)
```

Comment:
Looks great, thanks for catching the bug!

This fix will only work if args.main_file refers to a file in the current directory -- if args.main file is `examples/calculator.py`, importlib wouldn't be able to find the path. I think we can add the dirname to sys.path so that all imports in that path will be resolved directly. This has the advantage that we don't need to assume every folder in the path is a Python package.
```
import sys
...
sys.path.append(str(main_file.parent))
main_module = importlib.import_module(main_file.stem)
```